doc/using-js-ipfs-in-the-browser.md

```diff
@@ -0,0 +1,70 @@
+Using JS IPFS in The Browser
+----------------------------
+
+The document describes how you can use JS IPFS in the browser. JS IPFS is the
+implementation of the IPFS protocal for browser. JS IPFS runs in a browser as a
+service worked, a web extenstion and node.js. The document describes how you can
+boost the speed and reduce the errors you face often because of limitation of
+js-ipfs.
+
+There are few different components for implementing JS IPFS in the browser. These
+are webrtc and websocket-star which will make it easier for implemented JS IPFS
+in the browser and boost the speed and reduce the errors.
+
+You can find the information present for enabling webRTC support for js-ipfs in
+the browser [here](https://github.com/ipfs/js-ipfs#how-to-enable-webrtc-support-for-js-ipfs-in-the-browser) and a different one is
+implementing with webrtc-star, you can find information [here](https://github.com/ipfs/js-ipfs#is-there-a-more-stable-alternative-to-webrtc-star-that-offers-a-similar-functionality).
+
+You can find information about running IPFS in the broswer [here](https://github.com/ipfs/js-ipfs#table-of-contents).
+
+Getting Data From IPFS
+-----------------------
+
+Using Javascript in the browser
+
+```js
+const node = new IPFS()
+
+node.once('ready', () => {
+  node.pipe('QmPChd2hVbrJ6bfo3WBcTW4iZnpHm8TEzWkLHmLpXhF68A', (err, data) => {
+    if (err) return console.error(err)
+
+    // convert Buffer back to string
+    console.log(data.toString())
+  })
+})
+```
+
+JS IPFS Restriction in Web Browser
+------------------------------------------
+
+- There is no DHT in js-ipfs. 
+
+  In browser a JS IPFS node is more restricted because of the same origin policy
+  that decrease the discover/connect with peers, external peers without rendezvous
+  and relay servers, delegating peer/content routing and preload servers for content
+  discoverablilty hard and erroneous even with swarm peers.
+
+- Unable in opening TCP connections even if one sets up delegating routing the 
+issue will persist for connecting most of peers in the swarm as described a
+little above.
+
+- Connection with websocket ports should be adhering with [secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) limitations.
+
+JS IPFS Best Practices
+----------------------
+
+- Run in fedrated model.
```

Comment:
@tapaswenipathak what do you mean by this?

Reply:
I don't remember now.